core/controllers/creator_dashboard.py

```diff
@@ -326,7 +326,7 @@ def _round_average_ratings(rating):
             })
         else:
             self.values.update({
-                'topic_summary_dicts': None
+                'topic_summary_dicts': []
```

Comment:
> How could this error be detected, I mean typescript doesn't know what's there in the backend?

Ah I didn't realize you could assign null to any type. I think that explains it (I would have thought something else in the frontend would error).

>> Also, I'm surprised there isn't a backend test coverage error for these lines. Any idea why?

> There was a test for checking the response when the new structures are not enabled.

Could you please update the existing test, or add a new one, so that this error would have been caught? In general, always adopt that mentality when bug-fixing, i.e. try to make it such that the problem can never happen again in the future. Thanks!

Reply:
> Could you please update the existing test, or add a new one, so that this error would have been caught? In general, always adopt that mentality when bug-fixing, i.e. try to make it such that the problem can never happen again in the future. Thanks!

I already did that.

Reply:
Oh yep, sorry, missed that. Thanks!